spinalcordtoolbox/reports/slice.py

```diff
@@ -183,21 +183,19 @@ def add_slice(matrix, i, column, patch):
         return matrix
 
     @staticmethod
-    def nan_fill(A):
-        """Interpolate NaN values with neighboring values in array (in-place)
-        If only NaNs, return an array of zeros.
+    def inf_nan_fill(A):
+        """Interpolate inf and NaN values with neighboring values in a 1D array, in-place.
+        If only inf and NaNs, fills the array with zeros.
         """
-        nans = np.isnan(A)
-        if ~np.any(nans):
-            return A
-        elif np.all(nans):
-            A[:] = np.zeros_like(A)
-            return A
-        xp = (~nans).ravel().nonzero()[0]
-        fp = A[~nans]
-        x = nans.ravel().nonzero()[0]
-        A[nans] = np.interp(x, xp, fp)
-        return A
+        valid = np.isfinite(A)
+        invalid = ~valid
+        if np.any(valid):
+            A[invalid] = np.interp(
+                np.nonzero(invalid)[0],
+                np.nonzero(valid)[0],
+                A[valid])
+        else:
+            A.fill(0)
```

Comment:
Minor minor nitpick, but it took me a sec to parse the `if np.any(valid)` check (since we're in a method that's checking for inf and nan values, so I was wondering why we were checking if there were any valid values).

Also, the case where the array is all `valid` is subtly hidden in the "any(valid)" case, even though nothing will happen?

So, I wonder if it might be a bit clearer if we did this instead:

```suggestion
        if np.all(valid):
            pass
        elif np.all(invalid)
            A.fill(0)
        else:
            A[invalid] = np.interp(
                np.nonzero(invalid)[0],
                np.nonzero(valid)[0],
                A[valid])
```

Reply:
I'm happy to switch the condition from `np.any(valid)` to `np.all(invalid)` for readability, but I'd rather not split off the `np.all(valid)` case:
- The `np.all(invalid)` case split is necessary, because `np.interp` will raise an exception in that case.
- However the `np.all(valid)` case is naturally (at least to my mathematician eyes), correctly and efficiently handled by `np.interp`. And fewer code branches means fewer potential corner cases, typically.

Reply:
Eh, how about this compromise?
https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/07dc159a8dedf136367fffb14a47f91372d1e9fb/spinalcordtoolbox/reports/slice.py#L192-L198

I'll let you resolve the conversation if you're ok with it.

Reply:
What you've done in the follow-up commits LGTM! :blush: